lib/schema-fs-loader.js

```diff
@@ -37,43 +40,112 @@ const loadSchema = (filepath, context, callback) => {
   });
 };
 
-const loadFiles = (files, context, callback) => {
-  const schemas = [];
-  files.unshift(domainsPath);
-  metasync.each(files, (filepath, callback) => {
-    loadSchema(filepath, context, (err, name, schema, source) => {
-      if (err) {
-        callback(err);
-        return;
+const getSortOrder = file => {
+  if (!file.includes('.')) {
+    return 0;
+  }
+  if (file.endsWith('.category')) {
+    return 1;
+  }
+  return 2;
+};
+
+const sorter = (a, b) => getSortOrder(b) - getSortOrder(a);
+
+const loadModuleSchema = (
+  filepath,
+  context,
+  schemas,
+  dirName,
+  isRoot,
+  callback
+) => {
+  loadSchema(filepath, context, (error, name, definition, source) => {
+    if (error) {
+      callback(error);
+      return;
+    }
+
+    const type = common.fileExt(filepath);
+    if (type === 'domains') {
+      schemas.domains.push({ name, definition });
+    } else if (type === 'category') {
+      Object.entries(definition).forEach(([key, value]) => {
+        if (extractDecorator(value) === 'Action') {
+          delete definition[key];
+          schemas.actions.push({
+            name: key,
+            category: name,
+            definition: value,
+          });
+        }
+      });
+      schemas.categories.push({ name, definition });
+    } else {
+      let [category, entity] = name.split('.');
+      if (!entity) {
+        entity = category;
+        category = isRoot ? null : dirName;
       }
-      schemas.push([name, schema, source]);
-      callback(null);
-    });
-  }, err => {
-    if (err) callback(err);
-    else callback(null, schemas);
+      const obj = { name: entity, category, definition };
+      if (type === 'form') {
+        schemas.forms.push(obj);
+      } else if (type === 'view') {
+        if (!definition.Base) {
+          definition.Base = obj.category;
+        }
+        schemas.views.push(obj);
+      } else if (type === 'action') {
+        schemas.actions.push(obj);
+      }
+    }
+
+    schemas.sources.push({ path: filepath, source });
+    callback(null);
   });
 };
 
 // Load schemas from directory
 //   dir - <string> | <null> directory to load schemas from
 //   context - <Object> object to be assigned to global during loading
+//   schemas - <Object> object to collect schemas to
+//         categories - <Object[]>
```

Comment:
Incorrect indentation here and below.
```suggestion
//     categories - <Object[]>
```